superset-frontend/src/dashboard/components/nativeFilters/ChartCustomization/ChartCustomizationForm.tsx

```diff
@@ -0,0 +1,1147 @@
+          ...currentFilters,
+          [item.id]: {},
+        },
+      });
+    }
+  }, [form, item.id]);
+
+  const fetchDatasetInfo = useCallback(async () => {
+    const formValues = form.getFieldValue('filters')?.[item.id] || {};
+    const dataset = formValues.dataset || customization.dataset;
+
+    if (!dataset) {
+      setMetrics([]);
+      return;
+    }
+
+    try {
+      let datasetId: number;
+      if (
+        typeof dataset === 'object' &&
+        dataset !== null &&
+        'value' in dataset
+      ) {
+        datasetId = Number((dataset as any).value);
+      } else {
+        datasetId = Number(dataset);
+      }
+      if (Number.isNaN(datasetId)) return;
+
+      const response = await fetch(`/api/v1/dataset/${datasetId}`);
+      const data = await response.json();
+
+      if (data?.result) {
+        const datasetDetails = {
+          id: data.result.id,
+          table_name: data.result.table_name,
+          schema: data.result.schema,
+          database: data.result.database,
+        };
+
+        setDatasetDetails(datasetDetails);
+
+        const currentFilters = form.getFieldValue('filters') || {};
+        const currentItemValues = currentFilters[item.id] || {};
+
+        if (
+          currentItemValues.dataset &&
+          typeof currentItemValues.dataset === 'string'
+        ) {
+          const enhancedDataset = {
+            value: Number(currentItemValues.dataset),
+            label: data.result.table_name,
+            table_name: data.result.table_name,
+            schema: data.result.schema,
+          };
+
+          form.setFieldsValue({
+            filters: {
+              ...currentFilters,
+              [item.id]: {
+                ...currentItemValues,
+                dataset: currentItemValues.dataset,
+                datasetInfo: enhancedDataset,
+              },
+            },
+          });
+
+          onUpdate({
+            ...item,
+            customization: {
+              ...customization,
+              dataset: currentItemValues.dataset,
+              datasetInfo: enhancedDataset,
+            },
+          });
+        }
+
+        if (data.result.metrics && data.result.metrics.length > 0) {
+          setMetrics(data.result.metrics);
+        } else {
+          setMetrics([]);
+        }
+      }
+    } catch (error) {
+      console.error('Error fetching dataset info:', error);
+      setMetrics([]);
+    }
+  }, [form, item.id, customization.dataset]);
+
+  const fetchDefaultValueData = useCallback(async () => {
+    const formValues = form.getFieldValue('filters')?.[item.id] || {};
+    const dataset = formValues.dataset || customization.dataset;
+    const column = formValues.column || customization.column;
+    const hasDefaultValue =
+      formValues.hasDefaultValue ?? customization.hasDefaultValue;
+    const isRequired = formValues.isRequired ?? customization.isRequired;
+
+    if (hasDefaultValue && !isRequired) {
+      ensureFilterSlot();
```

Comment:
It seems we are updating fields values multiple times on this function and just 2 lines later. Is that intended? Can we improve the code structure to DRY this up a little bit?